scripts/pathseq/wdl/pathseq_pipeline.wdl

```diff
@@ -20,179 +26,712 @@
+##   - frac_final_total : estimated fraction of reads in the final output
+##   - frac_qual_cpx_filtered : estimated fraction of reads removed by low-quality/low-complexity filtering
+##   - frac_host_filtered : estimated fraction of reads removed by host filtering
+##   - frac_dup_filtered : estimated fraction of reads removed by deduplication
+## - If downsample = true:
+##   - total_reads_if_avail : total number of reads in the original input BAM
 ##
 ########################################################################################################################
 
-task PathseqPipeline {
+workflow PathSeqPipeline {
 
-  # Inputs for this task
   String sample_name
-  File input_bam
+  File input_bam_or_cram
+  # Required if a cram
+  File? input_bam_or_cram_index
 
-  File kmer_file
-  File filter_bwa_image
-  File microbe_bwa_image
-  File microbe_fasta
-  File microbe_fasta_dict
-  File taxonomy_file
+  # Required if the input is a cram
+  File? cram_reference_fasta
+  File? cram_reference_fasta_index
+  File? cram_reference_dict
 
+  # Set to true if host aligned. WARNING: Results in loss of EBV reads if in the aligned reference.
   Boolean is_host_aligned
-  Boolean? skip_quality_filters
+
+  File? kmer_file
+  File? filter_bwa_image
+  # Required if filtering_only=true
+  File? microbe_bwa_image
+  File? microbe_dict
+  File? taxonomy_file
+
+  # If enabled, filter metrics will be estimated using a downsampled bam with this many reads (recommended)
+  # If disabled, no filter metrics will be generated
+  Boolean downsample = false
+  Int downsample_reads = 1000000
+
+  # Enable to only perform host filtering
+  Boolean filtering_only = false
+
+  # Only recommended if downsample = true
+  Boolean gather_filter_metrics = false
+
+  # This can be calculated from a downsampled run to help optimize disk allocation during filtering
+  Float frac_non_host_reads = 1.0
+
+  # Filtering options
   Boolean? filter_duplicates
+  Boolean? skip_quality_filters
+  Int? host_min_identity
+  Int? host_kmer_threshold
+  Int? filter_bwa_seed_length
+  Int? min_clipped_read_length
+  Int? max_masked_bases
+  Int? min_base_quality
+  Int? quality_threshold
+  Int? dust_mask_quality
+  Int? dust_window
+  Float? dust_t
+  Int? max_adapter_mistmatches
+  Int? min_adapter_length
+  Int? filter_reads_per_partition
+  Int? filter_bam_partition_size
   Boolean? skip_pre_bwa_repartition
+
+  # Alignment options
+  Int? microbe_min_seed_length
+  Int? max_alternate_hits
+  Int? bwa_score_threshold
+
+  # Taxonomy scoring options
   Boolean? divide_by_genome_length
-  Int? filter_bwa_seed_length
+  Float? min_score_identity
+  Float? identity_margin
+  Boolean? not_normalized_by_kingdom
+  Int? score_reads_per_partition_estimate
+
+  # Runtime parameters
+  String gatk_docker
+  String samtools_docker = "biocontainers/samtools:v1.9-4-deb_cv1"
+  String genomes_in_the_cloud_docker = "broadinstitute/genomes-in-the-cloud:2.3.1-1512499786"
```

Comment:
I'm hoping to retire this repo.  What do you need out of there that's not in the GATK docker?  I think we have samtools and you should be able to call DownsampleSam via GATK.

Reply:
The version of samtools on the GATK docker is ancient and wasn't working for this task.

Reply:
@lbergelson any reason we can't rev our samtools?